tor/listen.go

```diff
@@ -116,6 +116,9 @@ type ListenConf struct {
 // If conf is nil, the default struct value is used. Note, if this errors, any
 // listeners created here are closed but if a LocalListener is provided it may remain open.
 func (t *Tor) Listen(ctx context.Context, conf *ListenConf) (*OnionService, error) {
+	if !conf.Version3 {
+		return nil, fmt.Errorf("v2 onion services no longer supported")
```

Comment:
What version are v2 hidden services removed on? Or is it the directory servers and what not that do not support them either? I don't want to break people using old Tor versions if it still works.

Also, if they are absolutely not supported anymore, I'd be ok just removing the field and all v2 related code and always assuming true.

Reply:
Ok, I see they are basically completely down on any Tor version. I think we should just remove this config option. I think a backwards incompatible change is acceptable here for removal of a field that is no longer respected.